### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by sindresorhus**

### CODE REVIEW

File: source/libs/page-detect.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable unicorn/prefer-starts-ends-with */
+/* eslint-disable unicorn/prefer-starts-ends-with, @typescript-eslint/prefer-string-starts-ends-with */
```
Comment (sindresorhus) [inline source/libs/page-detect.ts]: @bfred-it I didn't want to touch this as we already ignored a similar rule here, but I think we should fix this. Relying on `RegExp#test` coercing `undefined` to a string can be dangerous, as `undefined` might actually match something, causing a false-positive.

File: source/features/notification.d.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
 export interface Notification {
-	participants: { username: string | null; avatar: string }[];
+	participants: Array<{
+		username: string | null;
+		avatar: string;
+	}>;
```
Comment (bfred-it) [inline source/features/notification.d.ts]: Why `Array`?

File: source/libs/page-detect.ts (hunk 1 above)
Comment (bfred-it) [reply source/libs/page-detect.ts]: Ugh, you're right. `/undefined/.test(undefined) => true`

File: source/features/notification.d.ts (hunk 2 above)
Comment (bfred-it) [reply source/features/notification.d.ts]: Also, worth dropping that `null`?

File: source/features/mark-unread.tsx [hunk 3]
```diff
@@ -270,7 +270,7 @@ function isCurrentSingleRepoPage({repository}: Notification): boolean {
 }
 
 function isParticipatingPage(): boolean {
-	return /\/notifications\/participating/.test(location.pathname);
+	return location.pathname.includes('/notifications/participating');
```
Comment (bfred-it) [inline source/features/mark-unread.tsx]: Probably

```suggestion
	return location.pathname.startsWith('/notifications/participating');
```

File: source/features/notification.d.ts (hunk 2 above)
Comment (sindresorhus) [reply source/features/notification.d.ts]: It's more readable to only use `type[]` shorthand for simple types.

https://github.com/xojs/eslint-config-xo-typescript/commit/7b1f11891a4c2a3282792fcb139c7e134fe7f7dd

> Also, worth dropping that null?

Yes, but it's not really related to dependency updates.

File: source/features/mark-unread.tsx (hunk 3 above)
Comment (sindresorhus) [reply source/features/mark-unread.tsx]: Yup, it's `https://github.com/notifications/participating`.

File: source/features/notification.d.ts (hunk 2 above)
Comment (sindresorhus) [reply source/features/notification.d.ts]: Code compiles without the `null`, so I've removed it.

**Pull request merged**